pkg/cmd/openshift-kube-apiserver/server.go

```diff
@@ -0,0 +1,83 @@
+package openshift_kube_apiserver
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	utilwait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/pkg/version"
+	aggregatorinstall "k8s.io/kube-aggregator/pkg/apis/apiregistration/install"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/capabilities"
+	kubelettypes "k8s.io/kubernetes/pkg/kubelet/types"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/apis/config"
+	"github.com/openshift/origin/pkg/cmd/server/apis/config/validation"
+	"github.com/openshift/origin/pkg/cmd/server/origin"
+	"github.com/openshift/origin/pkg/cmd/util/variable"
+)
+
+func RunOpenShiftKubeAPIServerServer(masterConfig *configapi.MasterConfig) error {
+	// Allow privileged containers
+	capabilities.Initialize(capabilities.Capabilities{
+		AllowPrivileged: true,
+		PrivilegedSources: capabilities.PrivilegedSources{
+			HostNetworkSources: []string{kubelettypes.ApiserverSource, kubelettypes.FileSource},
+			HostPIDSources:     []string{kubelettypes.ApiserverSource, kubelettypes.FileSource},
+			HostIPCSources:     []string{kubelettypes.ApiserverSource, kubelettypes.FileSource},
+		},
+	})
+
+	if masterConfig.KubernetesMasterConfig == nil {
+		return fmt.Errorf("KubernetesMasterConfig is required to start this server - use of external Kubernetes is no longer supported.")
+	}
+
+	// install aggregator types into the scheme so that "normal" RESTOptionsGetters can work for us.
+	// done in Start() prior to doing any other initialization so we don't mutate the scheme after it is being used by clients in other goroutines.
+	// TODO: make scheme threadsafe and do this as part of aggregator config building
+	aggregatorinstall.Install(legacyscheme.GroupFactoryRegistry, legacyscheme.Registry, legacyscheme.Scheme)
```

Comment:
why is this special in our setup?

Reply:
> why is this special in our setup?

we enter after the command package upstream, so we need to make sure we add thsi.